sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -1073,7 +1093,7 @@ public Read<K, V> withRedistribute() {
     }
 
     public Read<K, V> withAllowDuplicates(Boolean allowDuplicates) {
-      if (!isAllowDuplicates()) {
+      if (!isRedistributed()) {
         LOG.warn("Setting this value without setting withRedistribute() will have no effect.");
```

Comment:
nit: seems lik ethis log should be printed when built instead of here. Since setWithAllowDuplicates could be called before setRedistributed

Reply:
Would this be best done in `KafkaIO.Read.expand()`? Looks like there is already a good amount of configuration checking done there. Or are you referring to somewhere else?

Reply:
Took a stab at it, please let me know if you had something else in mind.